RCTAppleHealthKit/RCTAppleHealthKit.m

```diff
@@ -329,4 +353,99 @@ - (void)getModuleInfo:(NSDictionary *)input callback:(RCTResponseSenderBlock)cal
+    int fiber = [[macros valueForKey:@"fiber"] integerValue];
+    int sugar = [[macros valueForKey:@"sugar"] integerValue];
+    int carbs = [[macros valueForKey:@"carbs"] integerValue];
+    int protein = [[macros valueForKey:@"protein"] integerValue];
+    
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.dateFormat = @"yyyy-MM-dd'T'HH:mm:ss.SSS";
+    NSDate *mealDate = [dateFormatter dateFromString:dateTimeString];
+    
+    NSDictionary *metadata = [[NSDictionary alloc] initWithObjectsAndKeys:id, @"mealId", name, @"recipe", type, @"type", nil];
+    
+    HKQuantityType *fatType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFatTotal];
+    HKQuantity *fatQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:fat];
+    HKQuantitySample *fatSample = [HKQuantitySample quantitySampleWithType:fatType quantity:fatQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *fiberType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFiber];
+    HKQuantity *fiberQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:fiber];
+    HKQuantitySample *fiberSample = [HKQuantitySample quantitySampleWithType:fiberType quantity:fiberQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *sugarType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietarySugar];
+    HKQuantity *sugarQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:sugar];
+    HKQuantitySample *sugarSample = [HKQuantitySample quantitySampleWithType:sugarType quantity:sugarQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *carbsType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCarbohydrates];
+    HKQuantity *carbsQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:carbs];
+    HKQuantitySample *carbsSample = [HKQuantitySample quantitySampleWithType:carbsType quantity:carbsQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *proteinType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryProtein];
+    HKQuantity *proteinQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:protein];
+    HKQuantitySample *proteinSample = [HKQuantitySample quantitySampleWithType:proteinType quantity:proteinQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *energyType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryEnergyConsumed];
+    HKQuantity *energyQuantity = [HKQuantity quantityWithUnit:[HKUnit kilocalorieUnit] doubleValue:energy];
+    HKQuantitySample *energySample = [HKQuantitySample quantitySampleWithType:energyType quantity:energyQuantity startDate:mealDate endDate:mealDate metadata:metadata];
```

Comment:
THis is a lot of repetition, and we could probs make a function out of it...but...not feeling too strongly about this